micro-rdk/src/common/provisioning/storage.rs

```diff
@@ -28,36 +42,68 @@ impl From<CloudConfig> for RobotCredentials {
     }
 }
 
-pub trait CredentialStorage {
+pub trait WifiCredentialStorage {
+    type Error;
+    fn has_wifi_credentials(&self) -> bool;
+    fn store_wifi_credentials(&self, creds: &WifiCredentials) -> Result<(), Self::Error>;
+    fn get_wifi_credentials(&self) -> Result<WifiCredentials, Self::Error>;
+}
+
+pub trait RobotCredentialStorage {
     type Error;
     fn has_stored_credentials(&self) -> bool;
     fn store_robot_credentials(&self, cfg: CloudConfig) -> Result<(), Self::Error>;
     fn get_robot_credentials(&self) -> Result<RobotCredentials, Self::Error>;
 }
 
+#[derive(Default)]
+struct MemoryCredentialStorageInner {
+    config: Option<RobotCredentials>,
+    ssid: Option<String>,
+    pwd: Option<String>,
+}
+
 /// Simple CrendentialStorage made for testing purposes
 #[derive(Default, Clone)]
-pub(crate) struct MemoryCredentialStorage {
-    config: Rc<Mutex<Option<RobotCredentials>>>,
-}
+pub(crate) struct MemoryCredentialStorage(Rc<Mutex<MemoryCredentialStorageInner>>);
 
-impl CredentialStorage for MemoryCredentialStorage {
+impl RobotCredentialStorage for MemoryCredentialStorage {
     type Error = Infallible;
     fn has_stored_credentials(&self) -> bool {
-        self.config.lock().unwrap().is_some()
+        let this = self.0.lock().unwrap();
```

Comment:
How exactly does `this` end up referencing the contained `MemoryCredentialsStorageInner` rather than being a value? I think, but am not sure, that it is because of `fn deref(&self) -> &T` on `MutexGuard`. Do I have that right? Maybe call it `inner_self` or `ref_inner` instead? Here and elsewhere if you take the suggestion or something similar.

Reply:
agreed :)